src/main/java/com/tstepnik/planner/service/StatisticsService.java

```diff
@@ -0,0 +1,55 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.Statistics;
+import com.tstepnik.planner.domain.User;
+import com.tstepnik.planner.repository.StatisticsRepository;
+import com.tstepnik.planner.repository.TaskRepository;
+import org.springframework.stereotype.Service;
+
+import java.text.DecimalFormat;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@Service
+public class StatisticsService {
+
+    TaskRepository taskRepository;
+    StatisticsRepository statisticsRepository;
+    TaskService taskService;
+    AuthService authService;
+
+    public StatisticsService(TaskRepository taskRepository, StatisticsRepository statisticsRepository,
+                             TaskService taskService, AuthService authService) {
+        this.taskRepository = taskRepository;
+        this.statisticsRepository = statisticsRepository;
+        this.taskService = taskService;
+        this.authService = authService;
+    }
+
+
+    public Statistics createAndGetStatistics() {
+        DecimalFormat df = new DecimalFormat("##.###");
+        User loggedUser = authService.getLoggedUser();
+        Integer userArchivedTasks = countArchivedTasks();
+        Integer userFinishedTasks = countFinishTasks();
+        Double userProductivity = (double)userFinishedTasks/ userArchivedTasks;
+        String format = df.format(userProductivity).replaceAll(",",".");
+        Double productivity = Double.valueOf(format);
+
+        return statisticsRepository.save(new Statistics(loggedUser.getId(),productivity, userArchivedTasks, userFinishedTasks));
+    }
+
+    public Integer countArchivedTasks() {
+        User loggedUser = authService.getLoggedUser();
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.of("UTC"));
+        Long numberOfTasks = taskRepository.countAllUserArchivedTasks(loggedUser.getId(),now);
+        return numberOfTasks.intValue();
+    }
+
+    public Integer countFinishTasks() {
+        User loggedUser = authService.getLoggedUser();
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.of("UTC"));
+        Long numberOfTasks = taskRepository.countAllUserFinishAndArchivedTasks(loggedUser.getId(),now);
```

Comment:
If method we are in are named `countFinishTasks` why are we getting finished and archived tasks? Is there a way that the task is done and not archived?

Reply:
yes, for example you finish task before expire date "planned for". I did that because through the day a lot of things can change and user can change his mind. So is better to count only archived tasks.

Reply:
I still don't get what you mean. This whole statistics seems a little bit more complicated because of a bunch of data that can change. It would be great to come up with the solution write it down and once we agree on it we can start coding it.

This method still doesn't make a lot of sense to me.

Reply:
https://github.com/tstepnik/planner/pull/32#discussion_r423784306

Reply:
Why are you linking this comment all the time? It doesn't help with that I said in my previous comment here.

Reply:
I link it only one time here. But if i change it, the main idea will be deprecated. If user didn't finish his task at the end of the day  he should not have acces to update task because it can provide to procrastination and he will keep postpone tasks :) According to complicated task it's really take as much power of DB ? Is so we can device another solution.